packages/base/src/util/getFastNavigationGroups.js

```diff
@@ -0,0 +1,65 @@
+let groups = [];
+
+const isFastNavGroup = $el => {
+	return $el.hasAttribute("data-sap-ui-fastnavgroup") && $el.getAttribute("data-sap-ui-fastnavgroup") === "true";
+};
+
+const isElementVisible = $el => {
+	const style = window.getComputedStyle($el);
+
+	return style.width !== "0px"
+		&& style.height !== "0px"
+		&& style.opacity !== "0"
+		&& style.display !== "none"
+		&& style.visibility !== "hidden";
+};
+
+const findFastNavigationGroups = (container, startFromContainer) => {
+	let child,
+		assignedElements,
+		index = 0;
+
+	if (!isElementVisible(container)) {
+		return;
+	}
+
+	if (isFastNavGroup(container)) {
+		groups.push(container);
+	}
+
+	if (container.shadowRoot) {
+		child = container.shadowRoot.firstChild;
+	} else if (container.assignedNodes && container.assignedNodes()) {
```

Comment:
instead of `conteiner.assignedNodes`, you could specifically check for shadow root to make it more obvious, but I don't mind, no need to change